src/impl/device/distanceSensor.cpp

```diff
@@ -0,0 +1,36 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+#include "okapi/impl/device/distanceSensor.hpp"
+
+namespace okapi {
+DistanceSensor::DistanceSensor(std::uint8_t iport, std::unique_ptr<Filter> ifilter)
+  : filter(std::move(ifilter)) {
+  port = iport;
+}
+
+DistanceSensor::~DistanceSensor() = default;
```

Comment:
The destructor, if it's default, should not be here and just marked as default in the header.